### PR TITLE
Properly reactivate PVT throwing

### LINKED ISSUE #2786: Modify compositionalMultiphaseWells tests and reactivate "Throw"
Merging https://github.com/GEOS-DEV/GEOS/pull/2552 triggered some integratedTests failure because it required new baselines and because some tests have pressure and temperature that go outside of the allowable bounds.

For now, I have fixed the issue in https://github.com/GEOS-DEV/GEOS/pull/2785 by replacing the `throw` with a warning. @dkachuma, @MelReyCG, we should review the tests that are failing, modify them appropriately, and then reactivate the `throw`.

The tests that are failing are:

````
FAIL RUN   : staircase_co2_wells_hybrid_3d_01                  : compositionalMultiphaseWell/staircase_co2_wells_hybrid_3d_01                       
FAIL RUN   : staircase_co2_wells_3d_08                         : compositionalMultiphaseWell/staircase_co2_wells_3d_08                             
FAIL RUN   : staircase_co2_wells_3d_01                         : compositionalMultiphaseWell/staircase_co2_wells_3d_01                       
FAIL RUN   : black_oil_wells_unsaturated_3d_04                 : compositionalMultiphaseWell/black_oil_wells_unsaturated_3d_04               
FAIL RUN   : black_oil_wells_unsaturated_3d_01                 : compositionalMultiphaseWell/black_oil_wells_unsaturated_3d_01                     
FAIL RUN   : black_oil_wells_saturated_3d_04                   : compositionalMultiphaseWell/black_oil_wells_saturated_3d_04                 
FAIL RUN   : black_oil_wells_saturated_3d_01                   : compositionalMultiphaseWell/black_oil_wells_saturated_3d_01                 
FAIL RUN   : PoroElastic_staircase_co2_3d_01                   : poroElasticCoupling/PoroElastic_staircase_co2_3d_01                         
FAIL RUN   : PoroElastic_staircase_co2_3d_04                   : poroElasticCoupling/PoroElastic_staircase_co2_3d_04                         
FAIL RUN   : deadOilEgg_smoke_3d_01                            : compositionalMultiphaseWell/deadOilEgg_smoke_3d_01                                
FAIL RUN   : staircase_co2_wells_hybrid_3d_08                  : compositionalMultiphaseWell/staircase_co2_wells_hybrid_3d_08

### PR DESCRIPTION
**Pull request by MelReyCG**
This PR aims to reactivate the exception throwing when a pressure or a temperature is outside of the PVT tables. The fix is to only do the check when `useSurfaceCondition` is `true` on the `WellControl` (the surface values that were sent when this option was set to `false` were set to a default `0` value, which were obviously outside of the table ranges).

According my tests, none of the tests that were mention the issue #2786 are still failing, but I couldn't verify that within the ATS environment, so the integrated tests must be run on this PR.

As requested by @dkachuma, the test is now done in `CompositionalMultiphaseWell::validateWellConstraints()`, there are now less `MultiFluidBase::checkTablesParameters()` calls (1 at init + 1 per timesteps), and the checks are still working on the case that was originally problematic (issue #2393).

### CODE REVIEW

File: src/coreComponents/physicsSolvers/fluidFlow/wells/CompositionalMultiphaseWell.cpp [hunk 1]
```diff
@@ -434,10 +432,29 @@ void CompositionalMultiphaseWell::validateWellConstraints( real64 const & time_n
...
-                 InputError );
+  if( wellControls.useSurfaceConditions() )
+  {
+    real64 const & surfaceTemp = wellControls.getSurfaceTemperature();
+    real64 const & surfacePres = wellControls.getSurfacePressure();
+
+    GEOS_THROW_IF( surfaceTemp <= 0,
+                   "WellControls " << wellControls.getDataContext() <<
+                   ": Surface temperature must be set to a strictly positive value",
+                   InputError );
```
Comment (dkachuma) [inline src/coreComponents/physicsSolvers/fluidFlow/wells/CompositionalMultiphaseWell.cpp]: Can we check the pressure here as well.

**Review by dkachuma — COMMENTED** [review]
Actually I made a mistake on this one:

- The check on surface pressure already exists in `WellControls.cpp` so there is no need to check here.
- The check on the surface temperature i.e. that `surfaceTemp > 0` is not in WellControls. I believe the issue is that the temperature might be irrelevant to some fluid models such as black oil models. So I would remove it from here. The `checkPVTTables` will then be the relevant check for the temperature depending on the fluid model.
- We need to do this check only once. I think `CompositionalMultiphaseWell::validateConstitutiveModels` might be the place.

**Comment by MelReyCG** [comment]
@dkachuma What do you think of `validateInjectionStreams()`? If it doesn't fit, I would prefer to add a dedicated method with the same signature as `validateInjectionStreams()`.

**Comment by dkachuma** [comment]
This looks good. Let me verify on a couple of the failed tests.

**Review by dkachuma — APPROVED** [review]
Looks good

**Comment by MelReyCG** [comment]
To keep track of the integratedTest run on PRs, maybe we should add a `ci: integratedTests passed` tag ?

**Comment by wrtobin** [comment]
This usage still looks appropriate @rrsettgast, but pending the discussion you initiated in https://fcmaelstrom.slack.com/archives/C03U0RPR5K2/p1701152915731069 we may want to refactor from try-catch-print to rcode-print-fail

**Comment by MelReyCG** [comment]
@wrtobin As the discussion evolved, can't we just merge this PR?
Anyway, if the try-catch is simply banned, we will need to refactor the code at many other places, `GEOS_THROW()` was already present before my work.

**Review by wrtobin — APPROVED** [review]

**Comment by jhuang2601** [comment]
This PR went through `ci: run integrated tests` and does NOT require rebaseline.
```
ATS END OF RUN STARTED December 08, 2023 20:54:18
    Status     : TestCase                              : Directory                                                 : Elapsed : Resources : TestStep  
    ---------- : ------------------------------------- : --------------------------------------------------------- : ------- : --------- : ----------
    FAIL CHECK : PoroElastic_permeableFault_smoke_01   : poroElasticCoupling/PoroElastic_permeableFault_smoke_01   : 0:00:05 : 0:00:05   : restartcheck
    FAIL CHECK : PoroElastic_impermeableFault_smoke_01 : poroElasticCoupling/PoroElastic_impermeableFault_smoke_01 : 0:00:05 : 0:00:05   : restartcheck
    ---------- : ------------------------------------- : --------------------------------------------------------- : ------- : --------- : ----------
```

**Pull request merged**